src/emulator/databaseEmulator.ts

```diff
@@ -173,12 +181,41 @@ export class DatabaseEmulator implements EmulatorInstance {
       if (e.context && e.context.body) {
         throw e.context.body.error;
       }
-      throw e.original;
+      if (e.original) {
+        throw e.original;
+      }
+      throw e;
     }
   }
 
-  private prettyPrintRulesError(filePath: string, error: string): string {
+  private prettyPrintRulesError(filePath: string, error: unknown): string {
```

Comment:
Optional: This includes some kinda complex type manipulation and casing. If you're up for it, it would be great to add some unit tests for this to make sure if behaves like we expect.

Reply:
TBH, we don't quite know what RTDB response format looks like so we'd be chasing a moving target anyway. Added a TODO comment though.